### PR TITLE
feature:gzip compressor

### PR DESCRIPTION
**Pull request by jsbxyyx**
### Ⅰ. Describe what this PR did
feature gzip compressor

### CODE REVIEW

**Comment by xingfudeshi** [comment]
@jsbxyyx CI is failed.

**Comment by jsbxyyx** [comment]
@xingfudeshi 1564 cause CI failed.

**Comment by l81893521** [comment]
I think if you can provide the unit test for CompressorFactory, that s better

File: core/src/main/java/io/seata/core/compressor/CompressorFactory.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        COMPRESSOR_MAP.put(CompressorType.NONE, new NoneCompressor());
+    }
+
+    /**
+     * Get compressor by code.
+     *
+     * @param code the code
+     * @return the compressor
+     */
+    public static synchronized Compressor getCompressor(byte code) {
```
Comment (zjinlei) [inline core/src/main/java/io/seata/core/compressor/CompressorFactory.java]: while use synchronized?

Comment (jsbxyyx) [reply core/src/main/java/io/seata/core/compressor/CompressorFactory.java]: avoid repeatedly load Compressor class.

**Pull request merged**